python/lsst/resources/http.py

```diff
@@ -221,6 +221,37 @@ def finalurl(r: requests.Response) -> str:
     return destination_url
 
 
+# Tuple (path, block_size) pointing to the location of a local directory
+# to save temporary files and the block size of the underlying file system
+_TMPDIR = None
+
+
+def _get_temp_dir() -> Tuple[str, int]:
+    """Return the temporary directory path and block size.
+
+    This function caches its results in _TMPDIR.
+    """
+    global _TMPDIR
+    if _TMPDIR:
+        return _TMPDIR
+
+    # Use the value of environment variables 'LSST_BUTLER_TMPDIR' or
+    # 'TMPDIR', if defined. Otherwise use current working directory
+    tmpdir = os.getcwd()
+    for dir in (os.getenv(v) for v in ("LSST_BUTLER_TMPDIR", "TMPDIR")):
```

Comment:
This package is nothing to do with Butler so it's not an obvious name. I know that `http.py` has plenty of other environment variables that do but I didn't change them because I was worried I would break your environment. I think this should probably be `LSST_RESOURCES_TMPDIR` (I do use a `LSST_RESOURCES_TEST_TMP` in tests but you wouldn't use that).